salt/cache/__init__.py

```diff
@@ -345,5 +345,10 @@ def store(self, bank, key, data):
         self.storage[(bank, key)] = [time.time(), data]
 
     def flush(self, bank, key=None):
-        self.storage.pop((bank, key), None)
+        if key is None:
+            for bank, key in tuple(self.storage):
+                if bank == bank:
+                    self.storage.pop((bank, key))
```

Comment:
```suggestion
            for _bank, _key in tuple(self.storage):
                if _bank == bank:
                    self.storage.pop((bank, key))
```
?
You masked the function `bank` variable

Reply:
:facepalm: nice catch

Reply:
Got this one now